### PR TITLE
cleanup eatcpu process after the test run

### PR DESCRIPTION
**Pull request by visheshh**
#### Describe Bug or Feature
eatcpu scripts keeps running after test finishes.

#### Describe Your Change
pgrep the pid of the eatcpu script and kill it.

#### Attach Test and Valgrind Logs/Output

[mainline_buggy.txt](https://github.com/openpbs/openpbs/files/5735549/mainline_buggy.txt)
[fixed_output.txt](https://github.com/openpbs/openpbs/files/5735560/fixed_output.txt)
mainline:

1094555 | vishv | 2022.0.0.20201221134055 | TestResourceUsageLog | vishesh | testdev-09-s12 | 0:08:27 | FAIL | 2020-12-23 02:04:40-08 | 5 | 1 | 0 | 0 | 0 | 0
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

Fixed:

1094547 | vishv | 2022.0.0.20201222134014 | TestResourceUsageLog | vishesh | testdev-09-s12 | 0:08:49 | FAIL | 2020-12-22 22:23:54-08 | 5 | 1 | 0 | 0 | 0 | 0
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

### CODE REVIEW

**Comment by birbalsain** [comment]
Changes are fine,.Please attech the logs

**Review by birbalsain — APPROVED** [review]
Changes LGTM

File: test/tests/functional/pbs_resource_usage_log.py [hunk 1]
```diff
@@ -55,18 +55,27 @@ def setUp(self):
         attr2 = {'resources_available.mem': '200gb'}
         self.server.manager(MGR_CMD_SET, NODE, attr2, id=self.mom.shortname)
 
+    def cleanup_eatcpu(self, scripts):
```
Comment (bhroam) [inline test/tests/functional/pbs_resource_usage_log.py:58]: Shouldn't the eat cpu script be cleaned up by the mom?  It's submitted as part of the job script.  If it isn't, that is a bug in mom, not the test.

File: test/tests/functional/pbs_resource_usage_log.py [hunk 2]
```diff
@@ -55,18 +55,27 @@ def setUp(self):
...
+        for script in scripts:
+            cmd = 'pgrep -f ' + script
+            ret = self.du.run_cmd(cmd=cmd, level=logging.DEBUG)
+            for pid in ret['out']:
+                cmd = 'sudo -u ' + str(TEST_USER) + ' kill -9 ' + pid
```
Comment (bhroam) [inline test/tests/functional/pbs_resource_usage_log.py:63]: No need for sudo.  Use the runas argument.

**Review by bhroam — CHANGES_REQUESTED** [review]

File: test/tests/functional/pbs_resource_usage_log.py (hunk 1 above)
Comment (visheshh) [reply test/tests/functional/pbs_resource_usage_log.py:58]: @bhroam You are correct. eatcpu script is cleaned up by mom. Only where the test case SIGKILL mom there the cleanup is not happening. So made changes to only those tests.

File: test/tests/functional/pbs_resource_usage_log.py (hunk 2 above)
Comment (visheshh) [reply test/tests/functional/pbs_resource_usage_log.py]: done

**Comment by visheshh** [comment]
Before fix 2 tests left the eatcpu process running.

pbsroot@machine:~/TEST/tmp/tests/functional> ps -ef | grep pbs
pbsuser    879     1 65  2020 ?        17-00:35:20 /opt/pbs/python/bin/python /tmp/PtlPbsp8y6ajhg.py
pbsuser    901     1 65  2020 ?        17-01:07:20 /opt/pbs/python/bin/python /tmp/PtlPbsg0ly7yb8.py
pbsuser    902     1 65  2020 ?        17-00:56:22 /opt/pbs/python/bin/python /tmp/PtlPbsg0ly7yb8.py

After fix test cleans it up,

**Comment by visheshh** [comment]
@bhroam Please review the changes.

**Review by bhroam — APPROVED** [review]
LGTM

**Pull request merged**